### DISCUSSION TITLE
ECS deploy runner git user error

### DISCUSSION THREAD

**Discussion by velarx** [gruntwork-io/knowledge-base#612] [How do I]
While post setup reference architecture, Gruntwork team shared the document for the post setup for gitlab CI, but that is giving the following error in the sanpshot. Please guide to setup ref arch and running. Error says "[ecs-deploy-runner][2022-12-08T11:22:35+0000] fatal: could not read Username for '[https://gitlab.xyz.com'](https://gitlab.xyz.com%27/): No such device or address".

Configuration were changed according for the VCS as well as GitHUBOAUTH.
Error:
![Screenshot_2](https://user-images.githubusercontent.com/91253270/206438454-50fa3b36-51cf-455a-98ec-5dfa7098007a.png)

Code local:
![image](https://user-images.githubusercontent.com/91253270/206438572-60df6311-71fd-4d43-a4f6-234c31c24537.png)

Code Planner and applier in _envcommon:
![image](https://user-images.githubusercontent.com/91253270/206438784-ddb42965-3dac-4675-954e-b958a3e00c09.png)

---

<ins datetime="2022-12-08T11:46:47Z">
  <p><a href="https://support.gruntwork.io/hc/requests/109707">Tracked in ticket #109707</a></p>
</ins>

**Comment by denis256** [comment]
Hello,
looks like the credentials to pull data from Gitlab aren't configured right, value from `gitlab_token_secrets_manager_arn` is plain Gitlab oauth2?

**Reply by velarx** [reply]
Hello,
For _envcommon .hcl file gitlab_token_secrets_manager_arn is the secret arn which is having the token VCSPAT token of the Gitlab. In shared ecsdeploy runner .hcl file also I had updated in the git config with the same secret arn.

It should be plain oauth2 for gitlab_token_secrets_manager_arn for gitlab? 
If so where should we provide the VCS pat in that case?

![image](https://user-images.githubusercontent.com/91253270/206826429-70e7580d-7656-488e-a6a9-b9ba69da66e9.png)
